File: src/Tracer.Shared/Authorization/Roles.cs
namespace Tracer.Shared.Authorization;

/// <summary>
/// Canonical role names used in seed data, JWT claims, and authorization checks.
/// </summary>
public static class Roles
{
    public const string SuperAdmin = "SuperAdmin";
    public const string SystemAdmin = "SystemAdmin";
    public const string ITAdmin = "ITAdmin";
    public const string AssetManager = "AssetManager";
    public const string InventoryManager = "InventoryManager";
    public const string ProcurementOfficer = "ProcurementOfficer";
    public const string DepartmentManager = "DepartmentManager";
    public const string FinanceOfficer = "FinanceOfficer";
    public const string Auditor = "Auditor";
    public const string HelpDesk = "HelpDesk";
    public const string Employee = "Employee";
    public const string Guest = "Guest";
}

File: src/Tracer.Web/src/app/core/auth/roles.ts
/**
 * Mirror of Tracer.Shared.Authorization.Roles — keep in sync with
 * src/Tracer.Shared/Authorization/Roles.cs
 */
export const Roles = {
  SuperAdmin: 'SuperAdmin',
  SystemAdmin: 'SystemAdmin',
  ITAdmin: 'ITAdmin',
  AssetManager: 'AssetManager',
  InventoryManager: 'InventoryManager',
  ProcurementOfficer: 'ProcurementOfficer',
  DepartmentManager: 'DepartmentManager',
  FinanceOfficer: 'FinanceOfficer',
  Auditor: 'Auditor',
  HelpDesk: 'HelpDesk',
  Employee: 'Employee',
  Guest: 'Guest',
} as const;

export type RoleName = (typeof Roles)[keyof typeof Roles];

/** Roles that use the personal (non-org-wide) dashboard. */
const END_USER_ROLES = new Set<string>([Roles.Employee, Roles.Guest]);

export function isEndUserRole(role: string | null | undefined): boolean {
  return !role || END_USER_ROLES.has(role);
}
